src/log.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from pathlib import Path
+
+
+def setup_logger(log_file: Path = None, level=logging.INFO) -> logging.Logger:
+    """
+    Sets up the logging configuration for the application.
+
+    Args:
+        log_file (Path): Optional path to a log file where logs will be saved.
+        level (int): Logging level (INFO, DEBUG, etc.).
+
+    Returns:
+        logging.Logger: Configured logger instance.
+    """
+    if log_file is not None:
+        log_file.parent.mkdir(exist_ok=True)
+
+    logger = logging.getLogger(__name__)
```

Comment:
this means that the logger will be `cala.log` (or `src.log` currently). maybe take logger name as an argument?

Reply:
oh crap, yeah i wanted it to be `cala.log`. i'll switch it out for an argument with a default.